src/sota_tools/server_credentials.cc

```diff
@@ -65,12 +65,13 @@ ServerCredentials::ServerCredentials(const boost::filesystem::path &credentials_
       if (strcmp(filename, "treehub.json") == 0) {
         json_stream = readArchiveFile(a);
         found_config = true;
-      } else if (strcmp(filename, "client.crt") == 0) {
-        client_cert_ = readArchiveFile(a)->str();
-      } else if (strcmp(filename, "client.key") == 0) {
-        client_key_ = readArchiveFile(a)->str();
-      } else if (strcmp(filename, "root.crt") == 0) {
-        root_cert_ = readArchiveFile(a)->str();
+      } else if (strcmp(filename, "client_auth.p12") == 0) {
+        client_p12_ = readArchiveFile(a)->str();
+        method_ = AuthMethod::kTls;
+      } else if (strcmp(filename, "api_gateway.url") == 0) {
+        use_api_gateway = true;
+        ostree_server_ = readArchiveFile(a)->str();
```

Comment:
I read this like `ostree_server` and `api_gateway` are the same entity.

Reply:
Unfortunately, for backwards compatibility, we have to support the case where there is no API gateway for some period. Going forward, there should always be such a gateway, but if we don't have the URL in the credentials.zip (or there wasn't such a gateway deployed), we still need to be able to talk to treehub directly.